src/tests/chatCompletions.test.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  getChatCompleteWithMessageContentArraysRequest,
+  getChatCompleteWithMessageStringRequest,
+} from './resources/requestTemplates';
+import transformToProviderRequest from '../services/transformToProviderRequest';
+import Providers from '../providers';
+import { Context, Hono, HonoRequest } from 'hono';
+import app from '../index';
+import providersConfig from './resources/providersConfig';
+import { CHAT_COMPLETIONS_ENDPOINT } from './resources/constants';
+import { createDefaultHeaders } from './resources/utils';
+
+for (const provider in providersConfig) {
```

Comment:
Instead of operation level, we can go by Provider Level. As we are aware of what each provider supports.